windows-gaming-driver/src/config.rs

```diff
@@ -3,20 +3,124 @@ use std::fs::{File, OpenOptions};
 use std::io::{Read, Write};
 
 use toml;
+use serde_yaml;
 
 #[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq, Copy)]
 pub struct DeviceId {
     pub vendor: u16,
     pub product: u16,
 }
 
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+pub enum UsbBinding {
+    ById(DeviceId),
+    ByPort { bus: u16, port: u16 },
```

Comment:
Create a new struct `ByPort` or `UsbPort` or so and make the variant `ByPort(UsbPoirt)`. While struct variants in enums are fine for matching, you can't write a function which takes it as argument and you can't use it in a generic context.